value/src/main/java/com/google/auto/value/processor/AutoValueProcessor.java

```diff
@@ -397,6 +387,26 @@ private void processType(TypeElement type) {
       }
     }
 
+    if (appliedExtensions.size() > 0) {
+      final Set<String> methodsToRemove = Sets.newHashSet();
+      for (int i = appliedExtensions.size() - 1; i >= 0; i--) {
+        AutoValueExtension extension = appliedExtensions.get(i);
+        Collection<String> consumed = extension.consumeProperties(context);
+        if (consumed != null) {
```

Comment:
I think it would be fine to require extensions to return a non-null set, which as @JakeWharton notes could be `Collections.emptySet()`. I don't think it's much of a burden for them to have to do that explicitly, though probably we should have `AutoValueExtension` be an abstract class so we can add methods to it (without using Java 8 default methods), and in that case obviously this could have a default implementation as described.